iceoryx_binding_c/include/iceoryx_binding_c/publisher.h

```diff
@@ -24,20 +25,33 @@
 /// @brief publisher handle
 typedef struct cpp2c_Publisher* iox_pub_t;
 
+/// @brief options to be set for a publisher
+typedef struct
+{
+    // size of the history chunk queue
+    uint64_t historyCapacity;
+
+    // name of the node the publisher belongs to
+    // nullptr indicates that the default node name is used
+    const char* nodeName;
+} iox_pub_options_t;
+
+/// @brief initialize publisher options to default values
+/// @param[in] options pointer to options to be initialized
```

Comment:
Sanity check makes sense. But what shall happen if it is not set? An output on std::err? Termination? The apocalypse? ... Questions, questions...

Reply:
On second thought, the sanity check variable is problematic since it must be public and therefore can confuse the user? What shall he do with it? I do not think it is worth it, wasted effort for not much gain.

Reply:
Use LogWarn now for uninitialized options.

Reply:
Sure, I would not terminate though but can add this as well. The reason is: if the user by accident or on purpose, sets all fields to a reasonable value, it will work.

We can of course not detect errors in ll cases, i.e. may have false negatives (but no false positives, so we would never terminate in a case were init was called unless the value was overwritten afterwards).

Reply:
The Problem is not that it might not work right now but that it works because everything is set manually and the user ignores the warning. If we add another option in half a year, there might be a weird behavior and the connection of the weird behavior to the warning might not be there, because it worked for half a year. So this is just to be future proof and force to user to write correct code as soon as possible.

Reply:
I understand this. Just wanted to make clear it is not entirely foolproof. And there are a lot of fools everywhere. ;-)